### ISSUE TITLE
docker composition - needs a reverse proxy

### ISSUE THREAD

**Issue by fvanderbiest** [georchestra/georchestra#1475]
As discussed earlier, (in https://github.com/georchestra/georchestra/pull/1474#issuecomment-246480763) I think that:
- the docker-master branch should be updated in order to have `georchestra.mydomain.org` as the default frontend FQDN
- the composition should include some kind of reverse proxy (eg https://hub.docker.com/r/jwilder/nginx-proxy/) sitting in front of our security-proxy, something like eg:

``` yml
rp:
  image: jwilder/nginx-proxy
  ports:
    - "80:80"
    - "443:443"
  volumes:
    - /var/run/docker.sock:/tmp/docker.sock:ro
    - /path/to/certs:/etc/nginx/certs

proxy:
  image: georchestra/security-proxy:latest
  ports:
    - "8080:8080"
  volumes:
    - /etc/georchestra:/etc/georchestra
  links:
    - database
    - ldap
    - cas
    - mapfishapp
    - ...
  environment:
    - VIRTUAL_HOST=georchestra.mydomain.org
    - VIRTUAL_PROTO=https
    - XMX=1g
```

**Comment by MaxiReglisse** [comment]
i agree to say that a reverse proxy in a container is useful when have a single georchestra instance. 
but what happens if you have several georchestra instances on the same docker host, georchestra-prod, georchestra-preprod and georchestra-dev ? i think a single reverse proxy is sufficient, which can be installed on the host.
you can also have other applications in other docker containers on the same host, and then a single reverse proxy is also sufficient. 
consequently, the reverse proxy should be optional.

**Comment by fvanderbiest** [comment]
True but you're an advanced user.  
Advanced users know how to configure their reverse proxies.
Newbies don't ...

**Comment by MaxiReglisse** [comment]
for the moment, i know how to configure an nginx reverse proxy in htttp, but not in https... ERR_SSL_PROTOCOL_ERROR. i will propose you something later.

**Comment by fvanderbiest** [comment]
Well, it should work out of the box with `jwilder/nginx-proxy` (provided it finds the certs he expects ...)

**Comment by fvanderbiest** [comment]
There's also https://github.com/containous/traefik to consider.

**Comment by MaxiReglisse** [comment]
yes yes yes ! :+1:

**Comment by fvanderbiest** [comment]
I gave it a try tonight, and it kind of works (see https://github.com/georchestra/georchestra/blob/traefik/docker-compose.yml#L18-L27 in branch traefik), but the SP keeps barking "Unknown host: georchestra.mydomain.org" in https://github.com/georchestra/georchestra/blob/traefik/security-proxy/src/main/java/org/georchestra/security/Proxy.java#L345-L352

Note that I also modified the cas and security-proxy properties file in my datadir to match the georchestra.mydomain.org fqdn.

```
proxy_1     | 2017-01-18 22:03:05.427:INFO:ROOT:qtp1865127310-17: org.tuckey.web.filters.urlrewrite.UrlRewriteFilter INFO: conf file modified since last load, reloading
proxy_1     | 2017-01-18 22:03:05.434:INFO:ROOT:qtp1865127310-17: org.tuckey.web.filters.urlrewrite.UrlRewriteFilter INFO: loaded (conf ok)
proxy_1     | 2017-01-18 22:03:06 security [ERROR] Unknown host: georchestra.mydomain.org
proxy_1     | 2017-01-18 22:03:06 statistics [INFO] anonymousUser|2017/01/18 22:03:06|http://header:8080/header/|psc|
proxy_1     | 2017-01-18 22:03:11 security [ERROR] Unknown host: georchestra.mydomain.org
proxy_1     | 2017-01-18 22:03:11 statistics [INFO] anonymousUser|2017/01/18 22:03:11|http://cas:8080/cas/login?service=http%3A%2F%2Fgeorchestra.mydomain.org%2Fj_spring_cas_security_check|psc|
proxy_1     | 2017-01-18 22:03:13 security [ERROR] Unknown host: georchestra.mydomain.org
proxy_1     | 2017-01-18 22:03:13 statistics [INFO] anonymousUser|2017/01/18 22:03:13|http://cas:8080/cas/css/cas.css;jsessionid=12omqbwrxq00g5vufexvarmr0|psc|
proxy_1     | 2017-01-18 22:03:13 security [ERROR] Unknown host: georchestra.mydomain.org
proxy_1     | 2017-01-18 22:03:13 statistics [INFO] anonymousUser|2017/01/18 22:03:13|http://cas:8080/cas/js/cas.js;jsessionid=12omqbwrxq00g5vufexvarmr0|psc|
proxy_1     | 2017-01-18 22:03:13 security [ERROR] Unknown host: georchestra.mydomain.org
proxy_1     | 2017-01-18 22:03:13 statistics [INFO] anonymousUser|2017/01/18 22:03:13|http://header:8080/header/|psc|
proxy_1     | 2017-01-18 22:03:22 security [ERROR] Unknown host: georchestra.mydomain.org
```

Any idea @pmauduit ?

**Comment by fvanderbiest** [comment]
traefik is highly interesting since:
 * it is highly customizable
 * it's feature-packed
 * it has an official docker image

**Comment by fvanderbiest** [comment]
> 2017-01-18 22:03:11 security [ERROR] Unknown host: georchestra.mydomain.org

Well that's "normal", since the SP container does not know how to resolve `georchestra.mydomain.org`, so `InetAddress.getByName` throws an error.

I fixed this with a1bbf62, taking advantage on docker-compose's internal DNS service:
```
root@815f042afdbd:/var/lib/jetty# ping georchestra.mydomain.org
PING georchestra.mydomain.org (172.20.0.2): 56 data bytes
64 bytes from 172.20.0.2: icmp_seq=0 ttl=64 time=0.107 ms
64 bytes from 172.20.0.2: icmp_seq=1 ttl=64 time=0.159 ms
64 bytes from 172.20.0.2: icmp_seq=2 ttl=64 time=0.140 ms
^C--- georchestra.mydomain.org ping statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max/stddev = 0.107/0.135/0.159/0.000 ms
```
